sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1154,7 +1196,8 @@ void CurlConnectionPool::MoveConnectionBackToPool(
 
   // Lock mutex to access connection pool. mutex is unlock as soon as lock is out of scope
   std::lock_guard<std::mutex> lock(CurlConnectionPool::ConnectionPoolMutex);
-  auto& hostPool = CurlConnectionPool::ConnectionPoolIndex[connection->GetHost()];
+  auto poolId = connection->GetHost() + connection->GetConnectionPropertiesKey();
```

Comment:
Why isn't the `host` included as part of the string returned from `GetConnectionPropertiesKey`? It seems odd to modify the return value again to get the actual pool ID.

Reply:
impl detail